modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -328,6 +331,100 @@ def dtype_builder(df):
     _index_cache = None
     _columns_cache = None
 
+    def set_index_cache(self, index):
+        """
+        Set index cache.
+
+        Parameters
+        ----------
+        index : sequence, callable or None
+        """
+        if isinstance(index, ModinIndex) or index is None:
+            self._index_cache = index
+        else:
+            self._index_cache = ModinIndex(index)
+
+    def set_columns_cache(self, columns):
+        """
+        Set columns cache.
+
+        Parameters
+        ----------
+        columns : sequence, callable or None
+        """
+        if isinstance(columns, ModinIndex) or columns is None:
+            self._columns_cache = columns
+        else:
+            self._columns_cache = ModinIndex(columns)
+
+    def has_index_cache(self):
```

Comment:
Should this be a property?

Reply:
Now all the methods you wrote about are properties.